agent/api/container/container.go

```diff
@@ -807,7 +813,15 @@ func (c *Container) MergeEnvironmentVariables(envVars map[string]string) {
 	}
 }
 
-func (c *Container) HasSecretAsEnv() bool {
+// Append the logging driver using secret into the container's LogConfig
+func (c *Container) ApplyLogDriverSecret(hostConfig *dockercontainer.HostConfig, logDriverSecretName string, logDriverSecretValue string) {
```

Comment:
i don't think this should be a method of container.. the container is not used. i think this can just be a helper function put below its caller